### PR TITLE
Processing duration tweak

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR tweaks the newly-added processing duration parameter so it _always_ returns values in milliseconds. The parameter definition's default unit guides which default unit type the widget wrapper should set itself too but will not impact on the widget's returned value unit (which is now always ms).

@nyalldawson , I've been haunted by this since my initial implementation. I think it's ultimately much better to stick to milliseconds only for the parameter values (easier to model, easier to use via qgis_process CLI, etc.). That constant will insure users don't have to go dig in each alg. what the unit type is set to.

### CODE REVIEW

**Comment by DelazJ** [comment]
I may be wrong but I think this affects the "stdbscan temporal" alg documentation ( i dont know if anything else), and if so ...

File: src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp [hunk 1]
```diff
@@ -1258,23 +1258,11 @@ QgsProcessingDurationParameterDefinitionWidget::QgsProcessingDurationParameterDe
   mDefaultLineEdit = new QLineEdit();
   vlayout->addWidget( mDefaultLineEdit );
 
-  vlayout->addWidget( new QLabel( tr( "Default unit type" ) ) );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:1261]: Why wouldn't we keep this?

Comment (nirvn) [reply src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:1261]: @nyalldawson , hm, was I feverish last night?! Reverting.

**Comment by nirvn** [comment]
@nyalldawson , comment addressed. Previous CI failure unrelated (the now infamous network manager test).

**Comment by nirvn** [comment]
@DelazJ , this shouldn't require a documentation change, the algorithm UI still offers the possibility to set duration in the temporal unit type of your choosing.

File: src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp [hunk 2]
```diff
@@ -1267,6 +1267,9 @@ QgsProcessingDurationParameterDefinitionWidget::QgsProcessingDurationParameterDe
   mUnitsCombo->addItem( QgsUnitTypes::toString( QgsUnitTypes::TemporalHours ), QgsUnitTypes::TemporalHours );
   mUnitsCombo->addItem( QgsUnitTypes::toString( QgsUnitTypes::TemporalDays ), QgsUnitTypes::TemporalDays );
   mUnitsCombo->addItem( QgsUnitTypes::toString( QgsUnitTypes::TemporalWeeks ), QgsUnitTypes::TemporalWeeks );
+  mUnitsCombo->addItem( QgsUnitTypes::toString( QgsUnitTypes::TemporalYears ), QgsUnitTypes::TemporalYears );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:1270]: I thought we intentionally omitted these because of the ambiguity in year length

Comment (nirvn) [reply src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:1270]: I was thinking about this again and thought it was still worth adding. For very large intervals, say a decade, its still useful.

Comment (nyalldawson) [reply src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp:1270]: Hmm - I still disagree. I think it's always going to be misleading, and if we only expose at most "days" then a user will quickly work out for themselves why the other units aren't an option..!

**Comment by nirvn** [comment]
Discussing this further with @nyalldawson , we've reached a good compromise: we keep the year, decade, century unit type, with an explicit suffix for the year label showing how many days it represents (365.25).

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**